src/interfaces/AEnemy.hpp

```diff
@@ -0,0 +1,69 @@
+#pragma once
+
+#include <AEntity.hpp>
+extern "C"
+{
+#include <raylib.h>
+}
+
+/// @brief Interfaz que representa el comportamiento de un enemigo en el juego.
+/// Los enemigos son entidades que atacan al jugador, se mueven con IA y
+/// sueltan botín al morir
+class AEnemy : public AEntity
+{
+protected:
+    /// @brief Cantidad de Pablo Coins que suelta al morir
+    int pabloCoinsAtDeath;
+
+    /// @brief Objetivo actual del enemigo (posición del jugador)
+    std::vector<Player> objectives;

```

Comment:
```suggestion
    std::vector<Player>& objectives;
```

Para que sea una referencia y no necesites pedir actualización todo el tiempo